src/Sylius/Behat/Context/Api/Admin/ManagingCustomerGroupsContext.php

```diff
@@ -23,17 +24,22 @@ final class ManagingCustomerGroupsContext implements Context
     /** @var ApiClientInterface */
     private $client;
 
-    public function __construct(ApiClientInterface $client)
+    /** @var ResponseCheckerInterface */
+    private $responseChecker;
+
+    public function __construct(ApiClientInterface $client, ResponseCheckerInterface $responseChecker)
     {
         $this->client = $client;
+        $this->client->setResource('customer_groups');
```

Comment:
Is the `$client` instance different for every context? Or is it shared?

If it's shared, there might be a case in which two context constructors will be called, so that only the last resource will stay on the client instance.

Reply:
Indeed 👌 Right not it's not a problem, as current API scenarios use steps only from one context 💃 I was thinking about not registering the ApiPlatformClient in the container explicitly but injecting to each context anonymous service with the $resource property already defined. What do you think about such an approach?

Reply:
Its good question, we can also add optional resource parameter in methods like `buildCreateRequest()` -> `buildCreateRequest(string resource = null)`, but this case make you to write more code for checking existing parameter

Reply:
Probably we can firstly go with multiple `ApiClient`s registered for each resource (with `$resource` property injected) and then refactor it to register them manually. WDYT?